### PR TITLE
ARROW-8784: [Rust] [DataFusion] Remove use of Arc from LogicalPlan

### PR DESCRIPTION
**Pull request by andygrove**
This PR replaces all uses of `Arc` with `Box` in the logical plan. It als removes a test that was still passing after this change but wasn't actually testing what it intended to test, and was therefore redundant.

### CODE REVIEW

**Review by nevi-me — COMMENTED** [review]
What's the benefit in replacing Arc with Box?

**Comment by andygrove** [comment]
> What's the benefit in replacing Arc with Box?

Box has less overhead than Arc but the reason for the change is really just that there was never a (good) reason to use Arc in the first place.

File: rust/datafusion/src/execution/context.rs [hunk 1]
```diff
@@ -233,11 +233,12 @@ impl ExecutionContext {
...
         match self.datasources.get(table_name) {
             Some(provider) => {
+                let schema = provider.schema().as_ref().clone();
                 let table_scan = LogicalPlan::TableScan {
                     schema_name: "".to_string(),
                     table_name: table_name.to_string(),
-                    table_schema: provider.schema().clone(),
-                    projected_schema: provider.schema().clone(),
+                    table_schema: Box::new(schema.to_owned()),
+                    projected_schema: Box::new(schema.to_owned()),
```
Comment (paddyhoran) [inline rust/datafusion/src/execution/context.rs:241]: Is the second `to_owned` needed here?

**Review by paddyhoran — APPROVED** [review]
LGTM

File: rust/datafusion/src/execution/context.rs (hunk 1 above)
Comment (andygrove) [reply rust/datafusion/src/execution/context.rs]: It wasn't. Thanks. I've removed it.

**Review by nevi-me — APPROVED** [review]

**Pull request closed without merging**